File: JUNE20A/TUPLE.py
# // Author: proRam
# // Name: Shubh Bansal
# // Dated: June 6, 2020
# // Question: https://www.codechef.com/JUNE20A/problems/TTUPLE

# /*
# Derivation 
# p * x1   +   y1    =    a
# q * x2   +   y2    =    b
# r * x3   +   y3    =    c
# */

import sys
sys.stdin = open('JUNE20A/input.txt', 'r') 
sys.stdout = open('JUNE20A/output.txt', 'w')

A,B,C,P,Q,R = 0,0,0,0,0,0

def ansis1( p,q,r,a,b,c ) :

    # // a = b = c 

    ans = 3 ;

    if ( a==b==c ) :
        if ( b==0 ) :
            return 1 ;
        else :
            ans = 2

    # // =================================================================

    # // p <---x----> a
    # // q .......... b
    # // r .......... c

    if ( q==b and r==c ) :
        return 1 ;

    # // =================================================================

    # // p <---x----> a
    # // q <---x----> b
    # // r .......... c

    # // try addition
    if ( a-p == b-q ) :
        if (r==c) :
            return 1;
        else :
            ans = 2

    # // try multiplication
    try :
        if ( p!=0 and q!=0 and a%p==0 and b%q==0 and a//p == b//q ) :
            if (r==c) :
                return 1;
            else :
                ans = 2
    except :
        pass

    # // =================================================================

    # // p <---x----> a
    # // q <---x----> b
    # // r <---x----> c    
    # // try addition
    if ( c-r == a-p and a-p == b-q ) :
        return 1 ;

    # // try multiplication
    try :
        if ( r!=0 and p!=0 and q!=0 and a%p==0 and b%q==0 and c%r == 0 and a//p == b//q and a//p == c//r ) :
            return 1 ;
    except :
        pass

    # // =================================================================

    return ans ;


def twoin1(p,a,q,b):
    if p==a or q==b :
        return True

    if a-p == b-q :
        return True

    try :
        if a%p==0 and b%q==0 and a//p==b//q :
            return True
    except :
        pass
	
    return False

def coincide(p,a,q,b,r,c) :
    x,d = b-a, q-p
    try :
        y = a - p*(x//d)
        if r * (x//d) + y == c :
            if ( x%d == 0 and (r+y==c or r*y==c) ):
                return True
    except :
        return False 


def postulate_eqsys( eqorder ) :

    global P,Q,R,A,B,C

    if eqorder == 0 :
        return (P,Q,R,A,B,C)
    elif eqorder == 1 :
        return (P,R,Q,A,C,B)
    elif eqorder == 2 :
        return (Q,P,R,B,A,C)
    elif eqorder == 3 :
        return (Q,R,P,B,C,A)
    elif eqorder == 4 :
        return (R,P,Q,C,A,B)
    elif eqorder == 5 :
        return (R,Q,P,C,B,A)


def main() :

    global P,Q,R,A,B,C

    for _ in range(int(input())) :
        P,Q,R = list(map(int,input().split()))
        A,B,C = list(map(int,input().split()))
        
        # try ans = 0
        if (P,Q,R)==(A,B,C) :
            print(0)
            continue 

        # Try ans = 1
        # if (P==A and twoin1(Q,B,R,C)) or (Q==B and twoin1(P,A,R,C)) or (R==C and twoin1(Q,B,P,A)) :
        #     print(1)
        #     continue
        
        # diff = A-P
        # if Q+diff==B and R+diff==C :
        #     print(1)
        #     continue
        # if P!=0 and A%P==0 :
        #     diff = A//P
        #     if  Q*diff==B and R*diff==C :
        #         print(1)
        #         continue
        
        # Try ans = 2
        

        for eqorder in range(6) :

            p,q,r,a,b,c = postulate_eqsys(eqorder)    

            a1 = ansis1( p,q,r,a,b,c )
            if ( a1 == 1 ) :
                print(1)
                break

            d = 0

            # 10
            # 11
            # 11
            if p!=0 and a%p==0 :
                d = a//p
                if twoin1(q*d,b,r*d,c) or twoin1(q*d,b,r,c) or twoin1(q,b,r*d,c) or twoin1(q,b,r,c) :
                    print('dtm')
                    print(2)
                    break
            
            d = a - p
            if twoin1(q+d,b,r+d,c) or twoin1(q+d,b,r,c) or twoin1(q,b,r+d,c) or twoin1(q,b,r,c) :
                print(2)
                break
        
            if coincide(p,a,q,b,r,c) :
                print(2)
                break

        else :

            print(3)

main()

'''
31
1 1 1
4 4 4
1 3 10
10000 3 18979
1 1 0
0 1 1
1 3 10
50 55 20
1 3 10
3 8 12
1 2 3
0 0 0
1 3 6
4 4 10
1 3 10
50 55 12
1 -3 7
5 6 8
2 3 5
7 11 13
5 10 13
53 103 133
1 3 -4
9 9 9
1 2 3
3 8 6
3 5 7
6 5 10
8 6 3
9 7 8
0 0 0
0 0 0
0 0 0
1 2 3
7 11 10
0 0 5
1 2 3
4 5 6
2 3 5
7 11 13
-15 7 1
-5  3 2
1 2 3
642812522 -800362287 63216506
479416287 -131377407 -485384906
-347797377 -31924853 174025552
-600464641 65936784 -404338570
-583978841 -41853773 -107403941
-115244701 -675856231 195451001
316770650 191379060 444940068
206615556 896617444 103019380
441909256 -657638099 -551749241
-407126904 -951344732 462148145
321213728 988318111 -289609678
891954866 908693161 588785979
-228359789 325816295 -786646575
-864898099 394695004 -914582447
-150731004 292399 738517704
-748650385 118117751 -819616428
-880331605 -153000204 590463802
-269398161 -824891562 -458682725
-471173780 -471435275 -678892583
'''

File: JUNE20A/TUPLE.cpp
// Author: proRam
// Name: Shubh Bansal
// Dated: June 6, 2020
// Question: https://www.codechef.com/JUNE20A/problems/TTUPLE

#include <bits/stdc++.h>
#pragma GCC target ("sse4.2")
using namespace std;
#ifndef ONLINE_JUDGE 
class Timer { 
    private: chrono::time_point <chrono::steady_clock> Begin, End; public: Timer () : Begin(), End (){ Begin = chrono::steady_clock::now(); } ~Timer () { End = chrono::steady_clock::now(); cerr << "\nDuration: " << ((chrono::duration <double>)(End - Begin)).count() << "s\n"; } } T;
#endif

using ll = long long ;
using ull = unsigned long long ;

struct Tuple {

    ll f,s,t ;
    // Tuple( ll x, ll y, ll z ) {
    //     f=x;
    //     s=y;
    //     t=z;
    // }

};

Tuple str,tar ;
int s2[] = {3,5,6}, s1[] = {1,2,4} ;


void select1( int i, ll &p, ll&q, ll&r, ll &a, ll&b, ll&c ) {

    if ( i==1 ) { 
        p = str.f ; a = tar.f ;
        q = str.s ; b = tar.s ;
        r = str.t ; c = tar.t ;
    } 
    else if ( i==2 ) { 
        q = str.f ; b = tar.f ;
        p = str.s ; a = tar.s ;
        r = str.t ; c = tar.t ;
    } 
    else { 
        r = str.f ; c = tar.f ;
        q = str.s ; b = tar.s ;
        p = str.t ; a = tar.t ;
    }
}

void select2( int i, ll &p, ll&q, ll&r, ll &a, ll&b, ll&c ) {

    if ( i==3 ) {
        p = str.s ;
        q = str.t ;
        r = str.f ;
        a = tar.s ;
        b = tar.t ;
        c = tar.f ;
    }
    else if ( i==5 ) {
        p = str.f ;
        q = str.t ;
        r = str.s ;
        a = tar.f ;
        b = tar.t ;
        c = tar.s ;
    }
    else {
        p = str.f ;
        q = str.s ;
        r = str.t ;
        a = tar.f ;
        b = tar.s ;
        c = tar.t ;
    }

}

int distinct( ll a, ll b, ll c ) {

    if ( a==b and b==c )
        return 1 ;
    if ( ( a==b and b!=c ) or ( a==c and c!=b ) or ( b==c and a!=b ) )
        return 2 ;

    return 3 ;

}

int32_t main(int argc, char** argv) {

    #ifndef ONLINE_JUDGE
    freopen("input.txt", "r", stdin) ;
    freopen("output.txt", "w", stdout) ;
    #endif
    ios_base::sync_with_stdio(false);
    cin.tie(nullptr) ; cout.tie(nullptr) ;

    int t ;
    ull tom ;
    cin >> t ;
    while ( t-- ) {

        cin >> str.f >> str.s >> str.t ;
        cin >> tar.f >> tar.s >> tar.t ;

        // CASE WHEN STR==TAR i.e. ans = 0
        if ( tar.f==str.f and tar.t==str.t and tar.s==str.s ) {
            cout << 0 << '\n' ;
            continue ;
        }

        // Derivation 
        // p * x1   +   y1    =    a
        // q * x2   +   y2    =    b
        // r * x3   +   y3    =    c

        ll P = str.f, Q = str.s, R = str.t ;
        ll A = tar.f, B = tar.s, C = tar.t ;
        ll X1 = (P!=0) ? A/P : 1, Y1 = (P!=0) ? A%P : A ;
        ll X2 = (Q!=0) ? B/Q : 1, Y2 = (Q!=0) ? B%Q : B ;
        ll X3 = (R!=0) ? C/R : 1, Y3 = (R!=0) ? C%R : C ;

        // cout << P << '*' << X1 << " + " << Y1 << " = " << A << endl ;
        // cout << Q << '*' << X2 << " + " << Y2 << " = " << B << endl ;
        // cout << R << '*' << X3 << " + " << Y3 << " = " << C << endl ;

        int d = (X1!=1)+(X2!=1)+(X3!=1)+(Y2!=0)+(Y2!=0)+(Y3!=0) ;


        cout << min(3,d -min(1,distinct(X1,X2,X3)-1) -min(1,distinct(Y1,Y2,Y3)-1) ) << '\n' ;

        if (1) {
        // int ans = 2 ;

        
        // // 1. 1 1 ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
        // if ( dx==1 and dy==1 ) {
        //     ans = (dx!=1)+(dy!=0) ;
        //     goto STOP ;
        // }

        // // 2. 1 2 ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
        // else if ( dx==1 and dy==2 ) {
        //     for ( int i : s2 ) {
        //         ll p=P,q=Q,r=R,a=A,b=B,c=C ;
        //         select2(i,p,q,r,a,b,c) ;
        //         ll x1 = (p!=0) ? a/p : 1, y1 = (p!=0) ? a%p : a ;
        //         ll x2 = (q!=0) ? b/q : 1, y2 = (q!=0) ? b%q : b ;
        //         ll x3 = (r!=0) ? c/r : 1, y3 = (r!=0) ? c%r : c ;
                
                
        //         // if ( y3 == 0 ) ans = 2 ;
        //         // if ( x1 == 1 ) ans = 2 ;
        //         // if ( y1 == 0 ) ans = 2 ;

        //         if ( x1==1 and y1==0 ) ans = 1 ; 
        //         if ( x1==1 and y3==0 ) ans = 1 ; 

        //         if ( ans==1 )
        //             goto STOP ;

        //     }
        // }

        // // // 3. 1 3 ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
        // // else if ( dx==1 and dy==3 ) {
        // //     for ( int i : s2 ) {
        // //         ll p=P,q=Q,r=R,a=A,b=B,c=C ;
        // //         select2(i,p,q,r,a,b,c) ;
        // //         ll x1 = (p!=0) ? a/p : 1, y1 = (p!=0) ? a%p : a ;
        // //         ll x2 = (q!=0) ? b/q : 1, y2 = (q!=0) ? b%q : b ;
        // //         ll x3 = (r!=0) ? c/r : 1, y3 = (r!=0) ? c%r : c ;
                
                
        // //         if ( x1 == 1 and y1==0 ) {
        // //             ans = 2 ;
        // //             goto STOP ;
        // //         }

        // //     }
        // // }

        // // 4. 2 1 ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
        // else if ( dx==2 and dy==1 ) {
        //     for ( int i : s2 ) {
        //         ll p=P,q=Q,r=R,a=A,b=B,c=C ;
        //         select2(i,p,q,r,a,b,c) ;
        //         ll x1 = (p!=0) ? a/p : 1, y1 = (p!=0) ? a%p : a ;
        //         ll x2 = (q!=0) ? b/q : 1, y2 = (q!=0) ? b%q : b ;
        //         ll x3 = (r!=0) ? c/r : 1, y3 = (r!=0) ? c%r : c ;
                
        //         // if ( x1 == 1 ) ans = 2 ;
        //         // if ( x3 == 1 ) ans = 2 ;
        //         // if ( y1 == 0 ) ans = 2 ;

        //         if ( x1==1 and y1==0 ) ans = 1 ; 
        //         if ( x3==1 and y1==0 ) ans = 1 ; 

        //         if ( ans==1 )
        //             goto STOP ;

        //     }
        // }

        // // // 7. 3 1 ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
        // // else if ( dx==2 and dy==1 ) {
        // //     for ( int i : s2 ) {
        // //         ll p=P,q=Q,r=R,a=A,b=B,c=C ;
        // //         select2(i,p,q,r,a,b,c) ;
        // //         ll x1 = (p!=0) ? a/p : 1, y1 = (p!=0) ? a%p : a ;
        // //         ll x2 = (q!=0) ? b/q : 1, y2 = (q!=0) ? b%q : b ;
        // //         ll x3 = (r!=0) ? c/r : 1, y3 = (r!=0) ? c%r : c ;

        // //         if ( x1==1 and y1==0 ) ans = 2 ; 

        // //         if ( ans==2 )
        // //             goto STOP ;
        // //     }
        // // }
    
        // // // 5. 2 2 ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
        // // else if ( dx==2 and dy==2 ) {

        // //     for ( int i : s2 ) {
        // //         ll p=P,q=Q,r=R,a=A,b=B,c=C ;
        // //         select2(i,p,q,r,a,b,c) ;
        // //         ll x1 = (p!=0) ? a/p : 1, y1 = (p!=0) ? a%p : a ;
        // //         ll x2 = (q!=0) ? b/q : 1, y2 = (q!=0) ? b%q : b ;
        // //         ll x3 = (r!=0) ? c/r : 1, y3 = (r!=0) ? c%r : c ;

        // //         if ( x1==x2 and x2!=x3 and y1!=y2 and y2==y3 ) {

        // //             if ( x3==1 and y2==0 ) ans = 2 ;
        // //             if ( x3==1 and y1==0 ) ans = 2 ;
        // //             if ( x1==1 and y2==0 ) ans = 2 ;
        // //             if ( x1==1 and y1==0 ) ans = 2 ;

        // //             if ( ans == 2 ) {
        // //                 goto STOP ;
        // //             }

        // //         }
                
        // //         else if ( x1==x2 and x2!=x3 and y1==y2 and y2!=y3 ) {

        // //             if ( x3==1 and y3==0 ) ans = 2 ;
        // //             if ( x3==1 and y1==0 ) ans = 2 ;
        // //             if ( x1==1 and y3==0 ) ans = 2 ;
        // //             if ( x1==1 and y1==0 ) ans = 2 ;

        // //             if ( ans == 2 ) {
        // //                 goto STOP ;
        // //             }

        // //         }

        // //     }
        // // }


        // // // 6. 2 3 ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
        // // // 8. 3 2
        // // // 9. 3 3
        // else if ( dx+dy>4 or (dx=1 and dy==3)  ) {
        //     ans = 3 ;
        //     goto STOP ;
        // }

        // // ----------------------------------------------------------------
        // STOP:
        // cout << ans << '\n' ;
        1;}
    }
    cout << endl ;

    return 0 ;   

}
